lib/Controller/MainController.php

```diff
@@ -218,6 +218,34 @@ public function category($category) {
 		}
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function categoryUpdate($category) {
+		$this->dbCacheService->triggerCheck();
+
+		$json = $this->restParser->getParameters();
+		if (!$json || !isset($json['name']) || !$json['name']) {
+			throw new Exception('New category name not found');
```

Comment:
Throwing a generic Exception is hard to catch later on. Also this will trigger a default request answer by NC that contains a stack trace and a human-readable HTML page to debug. Not ideal for AJAX requests.

As we are in the Controller here, I suggest creating a 4xx DataResponse here instead.

If you want to stick with the generic exception here, you should make it `\Exception` as we are in a namespace.

Reply:
updated to returning 400